### PR TITLE
llvmPackages.clang: move add-nostdlibinc-flag.patch to cc-wrapper

### PR DESCRIPTION
**Pull request by pwaller**
Remove a patch to decrease the divergence from upstream.

Motivation is a continuation of trying to reduce the number of clang
builds floating around. Wrapper builds are cheap, compiler builds are
not.

It's expected that this patch can be removed at some point and replaced
with some other cc-wrapper logic. This will be a little bit easier with
the logic already in the cc-wrapper.

Ongoing discussion in #191152. It would be nice to drop this but we
think it's probably needed at least on systems where /usr/include may be
available to prevent the possibility of picking up unwanted includes.

One potential method for eliminating nostdlibinc in the future is by
providing a non-existent -sysroot but that was reverted in #213185, and
remains a can of worms.

Signed-off-by: Peter Waller <p@pwaller.net>

## Things done

- Built on platform(s)
  - [ ] x86_64-linux
  - [ ] aarch64-linux
  - [ ] x86_64-darwin
  - [ ] aarch64-darwin
- For non-Linux: Is sandboxing enabled in `nix.conf`? (See [Nix manual](https://nixos.org/manual/nix/stable/command-ref/conf-file.html))
  - [ ] `sandbox = relaxed`
  - [ ] `sandbox = true`
- [ ] Tested, as applicable:
  - [NixOS test(s)](https://nixos.org/manual/nixos/unstable/index.html#sec-nixos-tests) (look inside [nixos/tests](https://github.com/NixOS/nixpkgs/blob/master/nixos/tests))
  - and/or [package tests](https://github.com/NixOS/nixpkgs/blob/master/pkgs/README.md#package-tests)
  - or, for functions and "core" functionality, tests in [lib/tests](https://github.com/NixOS/nixpkgs/blob/master/lib/tests) or [pkgs/test](https://github.com/NixOS/nixpkgs/blob/master/pkgs/test)
  - made sure NixOS tests are [linked](https://github.com/NixOS/nixpkgs/blob/master/pkgs/README.md#linking-nixos-module-tests-to-a-package) to the relevant packages
- [ ] Tested compilation of all packages that depend on this change using `nix-shell -p nixpkgs-review --run "nixpkgs-review rev HEAD"`. Note: all changes have to be committed, also see [nixpkgs-review usage](https://github.com/Mic92/nixpkgs-review#usage)
- [ ] Tested basic functionality of all binary files (usually in `./result/bin/`)
- [25.05 Release Notes](https://github.com/NixOS/nixpkgs/blob/master/nixos/doc/manual/release-notes/rl-2505.section.md) (or backporting [24.11](https://github.com/NixOS/nixpkgs/blob/master/nixos/doc/manual/release-notes/rl-2411.section.md) and [25.05](https://github.com/NixOS/nixpkgs/blob/master/nixos/doc/manual/release-notes/rl-2505.section.md) Release notes)
  - [ ] (Package updates) Added a release notes entry if the change is major or breaking
  - [ ] (Module updates) Added a release notes entry if the change is significant
  - [ ] (Module addition) Added a release notes entry if adding a new NixOS module
- [ ] Fits [CONTRIBUTING.md](https://github.com/NixOS/nixpkgs/blob/master/CONTRIBUTING.md).

---

Add a :+1: [reaction] to [pull requests you find important].

[reaction]: https://github.blog/2016-03-10-add-reactions-to-pull-requests-issues-and-comments/
[pull requests you find important]: https://github.com/NixOS/nixpkgs/pulls?q=is%3Aopen+sort%3Areactions-%2B1-desc

### CODE REVIEW

File: pkgs/build-support/cc-wrapper/default.nix [hunk 1]
```diff
@@ -629,6 +629,12 @@ stdenvNoCC.mkDerivation {
       echo " -L${libcxx_solib}" >> $out/nix-support/cc-ldflags
     ''
 
+    ## Prevent the compiler from seeing /usr/include. There is a desire to
+    # achieve this through alternate means. See discussion in NixOS/nixpkgs#191152.
```
Comment (emilazy) [inline pkgs/build-support/cc-wrapper/default.nix:633]: The Darwin‐related content from the comment above the conditional in the other file should move here (just the “on macOS … hopefully shouldn’t matter” bit should be fine).

**Comment by emilazy** [comment]
Given that apparently the `libc++` test case used to justify the LLVM patch previously now works without passing the flag at all (per discussion on Matrix), It would be good to verify if we actually need this. e.g., if we’re not passing it to GCC, is GCC currently leaky in the same way Clang would be without this flag? If so, then I don’t think it’s worth it to bother passing it at all, given the pitfalls.

cc @NixOS/llvm @NixOS/stdenv

File: pkgs/build-support/cc-wrapper/default.nix (hunk 1 above)
Comment (pwaller) [reply pkgs/build-support/cc-wrapper/default.nix]: Moved. Good catch, thanks!

Comment (emilazy) [reply pkgs/build-support/cc-wrapper/default.nix]: Still calls it a patch, but otherwise SGTM.

Comment (pwaller) [reply pkgs/build-support/cc-wrapper/default.nix]: s/skipping the patch/dropping the flag/

**Review by emilazy — APPROVED** [review]
Yay.

**Comment by pwaller** [comment]
Latest force removes extraneous blankline.

**Review by RossComputerGuy — APPROVED** [review]

**Comment by emilazy** [comment]
Successfully built `pkgsLLVM.hello` and `firefox` on `aarch64-linux`, I think this is good to go.

**Pull request merged**